### DISCUSSION TITLE
How to enable/activate ML feature's?

### DISCUSSION THREAD

**Discussion by aleem11** [deepgram/community#1009] [Feature Requests]
The parameters for sentiment, intent, and topic detection are not working for me. The Deepgram logs indicate that these parameters are not enabled for my account. How can I get these activated?  @michaeljolley @jcdyer

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The programming language you are working in (e.g. JavaScript, Python).
- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Reply by aleem11** [reply]
Programming language: The script is written in JavaScript.
Deepgram product: The script uses the Deepgram Live Transcription API for Speech to Text.

**Comment by jkroll-deepgram** [comment]
Hi @aleem11, can you please share your Deepgram project ID? If you log into the Deepgram Console, you'll see it listed at the top (and in the URL itself): like "Project ID: 00000000-0000-0000-0000-000000000000"

**Comment by aleem11** [comment]
My Deepgram project id from the console
is: 7f1bb92e-2fe1-482e-bcba-19c2a542d9d5

**Comment by jkroll-deepgram** [comment]
Hi @aleem11, thanks for sharing your project ID.

I don't see any permissions gating access to the ML (audio intelligence) features for your project.

If you see any errors when using them, can you please share the Deepgram request IDs and error messages?

Do note that these features are only available for English pre-recorded transcription, so you would see errors if requesting them on streaming transcription or non-English transcription.

**Comment by aleem11** [comment]
Thank you for your response. I tested the Deepgram sentiment feature again
with my project and confirmed that the sentiment feature is enabled.
However, I am not receiving sentiment data in the response, even though I
have included sentiment=true in my script.

To assist in troubleshooting, you can verify this in the last log data
using the following details:

   - *Request ID:* 3bd952f5-729d-42df-afa3-5653133ee8de
   - *Project ID:* 7f1bb92e-2fe1-482e-bcba-19c2a542d9d5

*Below is the complete response I received during my last test:*

json
{
  "type": "Results",
  "channel_index": [0, 1],
  "duration": 3.9499998,
  "start": 4.37,
  "is_final": true,
  "speech_final": false,
  "channel": {
    "alternatives": [
      {
        "transcript": "Hi, Krishna. Alim speaking here.",
        "confidence": 0.97314453,
        "words": [
          {
            "word": "hi",
            "start": 6.13,
            "end": 6.37,
            "confidence": 0.99365234,
            "speaker": 0,
            "punctuated_word": "Hi,"
          },
          {
            "word": "krishna",
            "start": 6.37,
            "end": 6.87,
            "confidence": 0.97314453,
            "speaker": 0,
            "punctuated_word": "Krishna."
          },
          {
            "word": "alim",
            "start": 7.0099998,
            "end": 7.17,
            "confidence": 0.87475586,
            "speaker": 0,
            "punctuated_word": "Alim"
          },
          {
            "word": "speaking",
            "start": 7.17,
            "end": 7.67,
            "confidence": 0.9951172,
            "speaker": 0,
            "punctuated_word": "speaking"
          },
          {
            "word": "here",
            "start": 7.8099995,
            "end": 8.309999,
            "confidence": 0.94189453,
            "speaker": 0,
            "punctuated_word": "here."
          }
        ]
      }
    ]
  },
  "metadata": {
    "request_id": "3bd952f5-729d-42df-afa3-5653133ee8de",
    "model_info": {
      "name": "2-general-nova",
      "version": "2024-01-18.26916",
      "arch": "nova-2"
    },
    "model_uuid": "c0d1a568-ce81-4fea-97e7-bd45cb1fdf3c"
  },
  "from_finalize": false}
*My code Snippet:
*
const params = {
      model: 'nova-2',
      language: 'en',
      smart_format: true,
      punctuate: true,
      paragraphs: true,
      diarize: true,
      sentiment: true,
    encoding: "mulaw",  // Example encoding
    sample_rate: 8000,  // Example sample rate

};

// Construct URL with parameters
const urlParams = new URLSearchParams(params).toString();
const deepgramUrl = `wss://api.deepgram.com/v1/listen?${urlParams}`;

**Comment by jkroll-deepgram** [comment]
Hi @aleem11, note that sentiment and other ML features are only available for [pre-recorded transcription](https://developers.deepgram.com/docs/getting-started-with-pre-recorded-audio). You're using [live streaming transcription](https://developers.deepgram.com/docs/getting-started-with-live-streaming-audio).

If you want to record your audio and then submit it to our [audio intelligence](https://developers.deepgram.com/docs/sentiment-analysis) REST API, or use streaming transcription and then send the text transcript to our [text intelligence endpoint](https://developers.deepgram.com/docs/text-sentiment-analysis), then you can get the sentiment information.
